src/columnar_transposition.rs

```diff
@@ -10,56 +10,80 @@ use common::{alphabet, keygen};
 use common::alphabet::Alphabet;
 
 /// A Columnar Transposition cipher.
-///
 /// This struct is created by the `new()` method. See its documentation for more.
 pub struct ColumnarTransposition {
     key: String,
+    null_char: Option<char>,
+    use_nulls: bool,
 }
 
 impl Cipher for ColumnarTransposition {
-    type Key = String;
+    type Key = (String, Option<char>);
     type Algorithm = ColumnarTransposition;
 
-    /// Initialize a Columnar Transposition cipher given a specific key.
+    /// Initialize a Columnar Transposition cipher given:
+    /// * a specific `key`, and
+    /// * an optional `null` - a character that will pad the columns
     ///
     /// Will return `Err` if one of the following conditions is detected:
     ///
     /// * The `key` length is 0.
     /// * The `key` contains non-alphanumeric symbols.
     /// * The `key` contains duplicate characters.
-    fn new(key: String) -> Result<ColumnarTransposition, &'static str> {
-        keygen::columnar_key(&key)?;
-        Ok(ColumnarTransposition { key: key })
+    /// * The `null` contains a character in the `key`
+    fn new(key: (String, Option<char>)) -> Result<ColumnarTransposition, &'static str> {
+        keygen::columnar_key(&key.0)?;
+
+        let use_nulls = match key.1 {
+            None => false,
+            Some(_c) => true,
+        };
+        let null_char = Some(key.1).unwrap();
+
+        if use_nulls && key.0.contains(null_char.unwrap()) {
+            return Err("The `null_char` cannot be be in the keyword.");
+        }
+
+        Ok(ColumnarTransposition {
+            key: key.0,
+            null_char: null_char,
+            use_nulls: use_nulls,
```

Comment:
I don't think I agree with storing the boolean `use_nulls`. The fact that the `null_char` is optional should be enough.